### PR TITLE
Added api implementation for tier resource

### PR DESCRIPTION
**Pull request by thilinicooray**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApi.java [hunk 1]
```diff
@@ -42,5 +42,83 @@ public Response tiersGet(@ApiParam(value = "Media types acceptable for the respo
     throws NotFoundException {
     return delegate.tiersGet(accept,ifNoneMatch);
     }
+    @POST
+
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApi.java]: Blank lines?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApiService.java [hunk 2]
```diff
@@ -17,5 +17,13 @@
 public abstract class TiersApiService {
     public abstract Response tiersGet(String accept,String ifNoneMatch)
     throws NotFoundException;
+    public abstract Response tiersPost(TierDTO body,String contentType)
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApiService.java]: Formatting issues

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/dto/TierDTO.java [hunk 3]
```diff
@@ -10,64 +12,132 @@
...
-    sb.append("  rate: ").append(rate).append("\n");
-    sb.append("  roles: ").append(roles).append("\n");
-    sb.append("}\n");
-    return sb.toString();
-  }
+
+
+    @NotNull
+    private String name = null;
+
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/dto/TierDTO.java]: Lines spaces no need

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApi.java (hunk 1 above)
Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApi.java]: Please note that these files get auto-generated using the swagger-to-cxf maven plugin we use for REST API development.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApiService.java (hunk 2 above)
Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/TiersApiService.java]: Please note that these files get auto-generated using the swagger-to-cxf maven plugin we use for REST API development.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/dto/TierDTO.java (hunk 3 above)
Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/gen/java/org/wso2/carbon/apimgt/rest/api/dto/TierDTO.java]: Please note that these files get auto-generated using the swagger-to-cxf maven plugin we use for REST API development.

**Pull request merged**